packages/screens/WebView/BrowserScreen.tsx

```diff
@@ -0,0 +1,380 @@
+import { useMemo, useState } from "react";
+import { SectionList, View, useWindowDimensions } from "react-native";
+import { useSelector } from "react-redux";
+
+import ListView from "../Mini/components/ListView";
+import MiniTextInput from "../Mini/components/MiniTextInput";
+
+import daoSVG from "@/assets/icons/dao-gray.svg";
+import governanceSVG from "@/assets/icons/governance-gray.svg";
+import loader from "@/assets/icons/loader.png";
+import marketSVG from "@/assets/icons/marketplace-gray.svg";
+import teritoriSVG from "@/assets/icons/networks/teritori.svg";
+import searchSVG from "@/assets/icons/search-gray.svg";
+import tnsSVG from "@/assets/icons/tns-gray.svg";
+import walletSVG from "@/assets/icons/wallet-grey.svg";
+import { BrandText } from "@/components/BrandText";
+import { SVGorImageIcon } from "@/components/SVG/SVGorImageIcon";
+import { ScreenContainer } from "@/components/ScreenContainer";
+import { SpacerRow } from "@/components/spacer";
+import { selectAvailableApps } from "@/store/slices/dapps-store";
+import { ScreenFC, useAppNavigation } from "@/utils/navigation";
+import {
+  neutral00,
+  neutral22,
+  neutral76,
+  neutral77,
+  withAlpha,
+} from "@/utils/style/colors";
+import { MOBILE_HEADER_HEIGHT, layout } from "@/utils/style/layout";
+import { dAppType } from "@/utils/types/dapp-store";
+
+const browserRoutes = [
+  {
+    routeName: "TNSHome",
+    route: "tns",
+  },
+  {
+    routeName: "NativeWallet",
+    route: "wallet-manager",
+  },
+  {
+    routeName: "Organizations",
+    route: "orgs",
+  },
+  {
+    routeName: "ToriPunks",
+    route: "dapp/tori-punks/welcome",
+  },
+];
+
+function WebViewHeader({
+  searchInput,
+  onChange,
+  hasNoDapps,
+}: {
+  searchInput: string;
+  onChange: (text: string) => void;
+  hasNoDapps: boolean;
+}) {
+  const navigation = useAppNavigation();
+
+  function onSubmit() {
+    if (hasNoDapps) {
+      const urlPattern: RegExp =
+        /(https:\/\/www\.|http:\/\/www\.|https:\/\/|http:\/\/)?[a-zA-Z]{2,}(\.[a-zA-Z]{2,})(\.[a-zA-Z]{2,})?\/[a-zA-Z0-9]{2,}|((https:\/\/www\.|http:\/\/www\.|https:\/\/|http:\/\/)?[a-zA-Z]{2,}(\.[a-zA-Z]{2,})(\.[a-zA-Z]{2,})?)|(https:\/\/www\.|http:\/\/www\.|https:\/\/|http:\/\/)?[a-zA-Z0-9]{2,}\.[a-zA-Z0-9]{2,}\.[a-zA-Z0-9]{2,}(\.[a-zA-Z0-9]{2,})?/g;
```

Comment:
we have an optimized reg ex in another PR, right?

Reply:
no, there is a fix commit in this PR